### ISSUE TITLE
AzureStorage seems to ignore OPTIONS

### ISSUE THREAD

**Issue by protoroto** [jschneier/django-storages#1470]
Using Django 4.2.16 and django-storages 1.14.4

I'm surely missing something here, but if I configure my Azure storage like this:

```
STORAGES = {
    "default": {
        "BACKEND": "storages.backends.azure_storage.AzureStorage",
        "OPTIONS": {
            "account_key": "<my_account_key>",
            "account_name": "<my_account_name>",
            "azure_container": "<my_container_name>",
        },
    },
}
```
it looks like the `OPTIONS` dictionary is ignored (I get authentication errors, missing container errors, and so on). 
If I ignore the options dict and set everything like:
```
AZURE_ACCOUNT_KEY = "<my_account_key>"
AZURE_ACCOUNT_NAME = "<my_account_name>"
AZURE_CONTAINER = "<my_container_name>"

STORAGES = {
    "default": {
        "BACKEND": "storages.backends.azure_storage.AzureStorage",
    },
}
```
everything works as expected. 
Do you have any ideas why this is happening?

**Comment by protoroto** [comment]
Furthermore, I can provide a test that demostrate this issue. The issue is related to this ticket https://code.djangoproject.com/ticket/35397 

It's working on Django 5.1, but failing on Django 5.0 and Django 4.2. See related pr #1471

**Comment by dimbleby** [comment]
> It's working on Django 5.1, but failing on Django 5.0 and Django 4.2. See related pr https://github.com/jschneier/django-storages/pull/1471

So this is working on recent django and you have identified the django bug that is responsible for that.  Is that right?

What do you think this repository should do about that?  You should simply use a version of django where this is not broken, no?

**Comment by protoroto** [comment]
@dimbleby I'm not expecting anything :) I was simply pointing out that what's written in this repository documentation is not valid, or at least it's not working for me. 
I'm not sure if it's the same bug, because on the Django ticket they're mentioning `override_settings`: I've opened this issue to see if I'm the only one affected by this or if I'm missing something.
If the documentation says that I should use `OPTIONS` dict to specify storage settings, and it's not working, what should I do? I'm totally up to try to solve this, if it's needed.

**Comment by dimbleby** [comment]
perhaps I am misunderstanding, I think you are saying that this works just fine with recent django

if that is so then you seem to be reporting an already-fixed bug in the wrong repository?

**Comment by protoroto** [comment]
@dimbleby I'm simply trying to understand if I'm doing something wrong here, or it's actually not working for everyone. Cases are two:
- I'm doing something wrong, so maybe someone can explain me what I'm doing wrong
- It's not working, so we need to change the documentation in this repository stating that with Django>=4.2 we can use `OPTIONS` dict
I'm fine with both options, and I'm willing to open a pr for updating the docs, but I'm not entirely sure that this problem is real or I've done something wrong.

**Comment by dimbleby** [comment]
IMO it is not up to django-storages to document bugs in django, but I will leave that to maintainers

**Comment by protoroto** [comment]
@dimbleby I totally agree with you! It's not up to django-storages to document bugs in Django! Problem is, I don't actually know if that's a Django bug, and I'm not entirely sure it's a bug at all. That's why I opened the issue asking for help, because maybe I was doing something wrong.

I've stumbled upon this only because I was trying to connect an Azure storage in a project using Django 4.2, and after following what's written here https://github.com/jschneier/django-storages/blob/master/docs/backends/azure.rst something wasn't working as expected. 

Then I started digging and trying and testing and it looked like what's written in this repository documentation wasn't right, and I've opened the issue.

I tried demostrating the issue with a test, and in my local setup the bug was showing only with Django 4.2 and Django 5.0, but not with Django 5.1, so I digged more and I've found the Django ticket, that maybe is explaining the problem.

I'm well aware that's not a django-storage issue, but maybe it could be useful for others stumbling upon this problem knowing that it's not working because of that.

And all of this is a big "maybe", because I've not seen anybody reporting this so maybe it's something I'm doing wrong and this is not a problem at all.
